src/Symfony/Component/Process/Process.php

```diff
@@ -268,20 +268,16 @@ public function start(callable $callback = null, array $env = array())
             }
         }
 
-        if (null === $env) {
-            $env = $this->env;
-        } elseif ($this->env) {
+        if ($this->env) {
             $env += $this->env;
         }
 
         $envBackup = array();
-        if (null !== $env) {
-            foreach ($env as $k => $v) {
-                $envBackup[$k] = getenv($k);
-                putenv(false === $v || null === $v ? $k : "$k=$v");
-            }
-            $env = null;
+        foreach ($env as $k => $v) {
+            $envBackup[$k] = getenv($k);
+            putenv(false === $v || null === $v ? $k : "$k=$v");
         }
+        $env = null;
```

Comment:
This is a bit weird to me. The http://php.net/proc_open function accept an array of the env var, and here we give `null`.
with `null` all the parent env is propagated, with `[]` nothing is propagated.

Then, With master, I don't know how to NOT inherit env var. Is it intended ?

Reply:
yes, that's what we planned for: inheriting happens all the time, and you need to unset explicitly if you don't want some env vars to not propagate

Reply:
Arf, that's not cool. How to I unset something ?
$
I tried
```
<?php

require __DIR__.'/vendor/autoload.php';

$process = new Symfony\Component\Process\Process('env');

$env = array_combine(array_values($_SERVER), array_fill(0, count($_SERVER), null));

$process->run(null, $env);

echo $process->getOutput();
```

But without success :/

Anyway this is issue is not related with the current PR. Should I open a ne ticket for that?

Reply:
> How to I unset something ?

set it to `false`
yes, please open a dedicated issue, with use case ;)